File: C++/2. Add Two Numbers.cpp
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode* l3 = NULL;
        ListNode* prep = NULL;
        ListNode* root = NULL;
        int c = 0;
        while(l1 != NULL && l2 != NULL){
            l3 = new ListNode((l1->val + l2->val + c) % 10) ;                
            c = (l1->val + l2->val + c) / 10;  
            if (root == NULL){                
                root = l3;
                prep = root;
            }
            else{                                
                prep->next = l3;                
                prep = l3;
            }
            l1 = l1->next;
            l2 = l2->next;
        }
        if (l1 == NULL && l2 != NULL){            
            while (l2 != NULL){                
                l3 = new ListNode((l2->val + c) % 10);
                c = (l2->val + c) / 10;
                prep->next = l3;
                prep = l3;
                l2 = l2->next;
            }
        }
        else if(l1 != NULL && l2 == NULL){
            while (l1 != NULL){                
                l3 = new ListNode((l1->val + c) % 10);
                c = (l1->val + c) / 10;
                prep->next = l3;
                prep = l3;
                l1 = l1->next;
            }
        }
        if (c != 0){
            l3 = new ListNode(c);                
            prep->next = l3;                                
        }
        return root;    
    }
};

File: Java/2. Add Two Numbers.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {        
        ListNode ans = new ListNode();
        ListNode head = ans;
        int cnt = 0;
        while (l1 != null || l2 != null) {
            int tmp = cnt;
            if (l1 != null) {
                tmp += l1.val;
                l1 = l1.next;
            }
            if (l2 != null) {
                tmp += l2.val;
                l2 = l2.next;
            }
            ans.next = new ListNode(tmp % 10);
            cnt = tmp / 10;
            ans = ans.next;
        }
        if (cnt != 0) {
            ans.next = new ListNode(cnt);
        }
        return head.next;
    }
}